parsing/ast_iterator.ml

```diff
@@ -89,6 +89,20 @@ let iter_opt f = function None -> () | Some x -> f x
 
 let iter_loc sub {loc; txt = _} = sub.location sub loc
 
+let rec iter_loc_lid sub lid =
+  let open Longident in
+  match lid with
+  | Lident _ -> ()
+  | Ldot (lid, id) ->
+    iter_loc sub lid; iter_loc_lid sub lid.txt; iter_loc sub id
+  | Lapply (lid, lid') ->
+    iter_loc sub lid; iter_loc_lid sub lid.txt;
+    iter_loc sub lid'; iter_loc_lid sub lid'.txt
+
+let iter_loc_lid sub {loc; txt} =
+  iter_loc_lid sub txt;
+  iter_loc sub {loc; txt}
```

Comment:
Sorry, I missed that point earlier, but we probably want to have `iter_loc` first to keep the iterator a depth-first traversal.